lib/browser/rpc.js

```diff
@@ -334,20 +358,12 @@ function sendRequest(command, data, host = sessionHost) {
             let error = response && response.error;
 
             // Remove the type prefix from the error message (e.g. "Error: ").
-            if (error && error.replace) {
-                error = error.replace(/^[a-z]+: /i, '');
-            }
-            else if (error.type && error.type === 'dict') {
-                const responseError = deserialize_json_value(error);
-                let responeMessage;
-                if (response.message && response.message !== '') {
-                    // Remove the type prefix from the error message (e.g. "Error: ").
-                    responeMessage = response.message.replace(/^[a-z]+: /i, '');
-                }
-
-                const exceptionToReport = new Error(responeMessage);
-                Object.assign(exceptionToReport, responseError);
-                throw exceptionToReport;
+            if (error && typeof error === "string") {
+                error = error.replace(/^[a-z]+: /i, "");
+            } else if (error.type && error.type === "error") {
```

Comment:
Just to confirm we are sure we don't have this case anymore?
`error.type && error.type === 'dict'`

Reply:
We should definitely refrain from throwing anything other than instances of `Error` (or a subclass thereof).
If we do,  that will proceed to line 369 - so an error will be thrown client side, although that will probably be with a strange message:

    Error: [object Object]

Reply:
aren't we throwing an instance of Error in the old implementation. Seems we are
https://github.com/realm/realm-js/pull/3411/files#diff-1aeca8a83cdc5d4f7bf6859366830e44c8acbd0269a540bef9194be798436902L348
I wonder if the case with this check is still valid
`error.type && error.type === 'dict'`

Reply:
> aren't we throwing an instance of Error in the old implementation.

We are, but that would be serialized to a 'dict' which deserialized to a regular object on the client side, which would then be wrapped in an Error.

> We should definitely refrain from throwing anything other than instances of Error (or a subclass thereof).

This comment was to the native code. It's the only way I can see the `error.type` would be "dict" after this PR.

Reply:
ok. resolving